prototypes/technology.lua

```diff
@@ -116,5 +120,167 @@ data:extend(
 		},
 		order = "c-k-b"
 	},
-})
-
+	--     -- TINY --
+	-- Warehouse Research
+	{
+		type = "technology",
+		name = "warehouse-research-tiny",
+		icon = "__Warehousing__/graphics/research/warehouse-research-tiny.png",
+		icon_size = 128,
+		effects =
+		{
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-basic-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-basic-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-basic-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-basic-tiny",
+			},
+		},
+		prerequisites = { "steel-processing" },
+		unit =
+		{
+			count = 100,
+			ingredients =
+			{
+				{ "automation-science-pack", 1 },
+				{ "logistic-science-pack", 1 },
+			},
+			time = 30
+		},
+		order = "c-a"
+	},
+	-- Logistics Warehouse Research 1
+	{
+		type = "technology",
+		name = "warehouse-logistics-research-1-tiny",
+		icon = "__Warehousing__/graphics/research/warehouse-logistics-research-tiny.png",
+		icon_size = 128,
+		effects =
+		{
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-passive-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-storage-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-passive-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-storage-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-passive-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-storage-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-passive-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-storage-tiny",
+			},
+		},
+		prerequisites = { "warehouse-research-tiny", "robotics" },
+		unit =
+		{
+			count = 300,
+			ingredients = {
+				{ "automation-science-pack", 1 },
+				{ "logistic-science-pack", 1 },
+			},
+			time = 30
+		},
+		order = "c-k-a"
+	},
+	-- Logistics Warehouse Research 2
+	{
+		type = "technology",
+		name = "warehouse-logistics-research-2-tiny",
```

Comment:
Have you tested these in-game? Do they get level numbers added automatically like the existing logistic warehousing techs do? If not, these new ones will probably have to be changed to _end_ with the number. (Sadly I can't test this myself at the moment.)

Reply:
Since you added a second layer of research to the latest version I have not tested it in game.  I probably won't be doing a play through until all the factorio patches settle down.  which given previous major releases is usually 2-4 weeks from release.  I did however load the game with my changes and it didn't crash or have an issue when giving the command to research all techs so I would assume it would work while actually doing the research.

Reply:
This PR is going to sit in review for long enough that you'll definitely have time to play-test these once you're ready to do so. We're already approaching two weeks since 0.17 dropped. :smile: